src/queue/queue.ts

```diff
@@ -0,0 +1,57 @@
+// import * as batch from "../batch/batch";
+import { getConfig, Config } from "../config/config";
+// import * as announcement from "../contracts/announcement";
+// import * as storage from "../storage/storage";
+import { DSNPMessage } from "../messages/messages";
+import { HexString } from "../types/Strings";
+import { NotImplementedError } from "../utilities";
+
+export type QueueId = HexString;
+
+export interface QueueInterface {
+  enqueue(dsnpMessage: DSNPMessage): Promise<QueueId>;
+  dequeue(id: QueueId): Promise<DSNPMessage>;
+  getAll(): Promise<DSNPMessage[]>;
```

Comment:
peak?

Reply:
I scrapped `getAll()` in the latest re-write. `peek()` might be a good addition, but at the moment, I'm trying to keep the interface as minimal as possible as to not accidentally exclude any future queue adapters.